config/initializers/002-rails_failover.rb

```diff
@@ -36,19 +36,34 @@
     end
   end
 
-  RailsFailover::ActiveRecord.on_failover do
-    if RailsMultisite::ConnectionManagement.current_db == RailsMultisite::ConnectionManagement::DEFAULT
+  RailsFailover::ActiveRecord.on_failover do |role|
+    if role == ActiveRecord::Base.writing_role # Multisite master
       RailsMultisite::ConnectionManagement.each_connection do
-        Sidekiq.pause!("pg_failover") if !Sidekiq.paused?
         Discourse.enable_readonly_mode(Discourse::PG_READONLY_MODE_KEY)
       end
+    else
+      ActiveRecord::Base.connected_to(role: role) do
```

Comment:
I can't remember clearly but just to confirm this will be thread safe?

Reply:
Yes I think so. It [calls `ActiveRecord::Base.connection_handler=`](https://github.com/rails/rails/blob/53331b478d32e42a8155414379b2edbe4277a3a4/activerecord/lib/active_record/connection_handling.rb#L390-L395
), which uses [a thread-local variable](https://github.com/rails/rails/blob/53331b478d32e42a8155414379b2edbe4277a3a4/activerecord/lib/active_record/core.rb#L160-L166)